### DISCUSSION TITLE
NO reed switch - blue LED stops flashing on regaining contact

### DISCUSSION THREAD

**Discussion by kotylo** [PricelessToolkit/MailBoxGuard#36] [Q&A] [answered]
https://github.com/PricelessToolkit/MailBoxGuard/assets/2077668/64c10a75-977a-4ebf-9484-e85e5b651174

So, i wanted to install it finally in my mailbox, but found another weird behavior. I have the NO reed switch. So the current will only flow in the presence of the magnet. As i imagine, it should always be active, except when the mailbox slot is slid open to throw some mail (for 1-3 seconds usually, i imagine). Once magnet goes far away, the led starts flashing (booting?), but if i don't give it enough time (10 seconds?) then the magnets are back on place and the flashing stops, the LoRa packet is not sent. Is that expected behavior?

**Comment by PricelessToolkit** [comment]
The board will wake up only when the reed switch goes from "close to open". Led should not blink.

I hope that the firmware is uploaded and the right Attiny is selected.
1. Check if the "reed switch" can disconnect contact properly (don't use too powerful magnet)
2. Check values of R1 100K, R10 220K and C1 10uF

if all the values are correct then the only thing left is to check the connection between ( LATCH 'Pin8' on Attiny ) and (Q1-Base).

**Reply by kotylo** [reply]
Thanks for helping (again!)
Reed switch seems to disconnect properly with my magnets, tested via continuity.

I could only find ATTINY1616-M**N**R, instead of needed "-MFR". The difference that i've compared was the speed:
- MNR: 20MHz
- MFR: 16MHz

I've checked the flashing parameters from your youtube video and it was running @ 5Mhz anyway, so should not make a difference, i hope :) It sends the two packets anyway, but only on solid blue LED (at the end of my video above)

Admittedly, R10 was the only resistor to which i didn't have the exact 220K value, placed 200K one. 
Measured on PCB (without desoldering): R1 98K✔, C1 13uF✔, R1 150K ❌ (maybe it's influenced by other components).

For R10, i'll try to desolder it and replace with 200K + 20K in series, but I have my doubts if that's important :)
And I'll also try the connection check you mentioned.

**Reply by PricelessToolkit** [reply]
Power ON time until Attiny boots up and enables the latch pin,  is determined by R10.

**Reply by kotylo** [reply]
1. Not sure how to test connection from pin8 to Q1 base, but that worked: [PIN8]-[R3], [D1 cathode]-[Q1 base]. So D1 was not tested.
2. Desoldered C1 just to make sure i measure outside of PCB: 16uF ✔
3. Soldered R10(200K+18K) in series, was fun, but still the same behavior.
4. I've noticed that C5 makes "clicking" sounds (placed tweezers to cap, another end to my ear) in tact with the LED flashes. Should i bother with testing it or that can happen? :)

**Reply by PricelessToolkit** [reply]
You can trigger the board and test the C5 with a multimeter "if it's a fast one" and see the voltage, it's not normal that the C5 is clicking, maybe the voltage is fluctuating with the (LED in "sync" ? .) and 16uF is not 10uF so ❌ :)

**Reply by kotylo** [reply]
Yes, i meant LED in sync, sorry.

> 16uF is not 10uF so ❌

Oh boy, i thought that it's like electrolytical capacitor, if a bit larger, than not that important. Seems like my aliexpress ceramic capacitor kit shows it's quality, though. The 10 ones are actually 14.5, 14.7, 15.5. I see it is supposed to be within 10% range, so they must have mixed it up with 15uF ones :( ... Replaced with the lowest found: 14uF, still the same behavior. Is it critical here to be 10, though? i mean i could try to find just 10uF ones.
p.s. The 22uF C5 I got from LCSC, it is point on 22.1uF and much better quality, so my multimeter is not at fault.

> test the C5 with a multimeter "if it's a fast one"

i feel like i'm getting deeper into unknown territory. If you'll have some time, could you guide me in the right direction, how to do it? I've tried searching, but can't seem to find any meaningful information

**Reply by PricelessToolkit** [reply]
14 is ok, just test the voltage with a multimeter on C5 when the LED is pulsing it will fluctuate from 3.20-3.30V, not 2~3V.

**Reply by kotylo** [reply]
1.0-1.1V while flashing, then 3.3 while solid LED, LoRa sent.

**Reply by PricelessToolkit** [reply]
it switches off the trigger circuit mosfet too soon, for that reason Attiny can't boot up for "self" powering.  let me think about it, and good night.

**Reply by kotylo** [reply]
thanks a bunch, good night 🤗

**Reply by kotylo** [reply, marked as answer]
i was talking to ChatGPT, i know, i know 😆. But it had idea of checking with stable power supply. So i did with 3V power bench. The LED didn't flash once, but dimmed a bit. But it mostly worked, if i gave it time between tries. So the next option was to use another battery. I did that, and the flashing stopped (114 mOhm internal resistance). Dimming too. I didn't think of battery issues before, because:
- [soldered] my 3-year-old 500mA LiPo battery was never used I thought it should work ok because the current consumption is low here. Maybe it's quality of the battery has decreased over time. Charged to 90%.
  - 440 mOhm internal resistance
- [connected via neodymium magnets] my 10 year old (now I know, i should not try to find use of my old laptop 18650s now) LiIon battery also had the same behavior. So as another one. And it was fully charged.
  - 130-150mOhm internal resistance.

I guess check you batteries. Maybe some internal resistance is high or just bad contact, but they still do work in a flashlight. I think the question is resolved then :) and again a "dumb" question from my side, i guess.

**Reply by PricelessToolkit** [reply]
I also didn't think that this could be due to a dead battery, I already thought it was an LDO problem. I talk to ChatGPT too :)
